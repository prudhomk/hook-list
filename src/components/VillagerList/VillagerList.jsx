import React from 'react';
import PropTypes from 'prop-types';
import Villager from '../Villager/Villager';

const VillagerList = ({ villagers }) => {
  const villagerElements = villagers.map((villager) => (
    <li key={villager._id}>
      <Villager
        {...villager}
      />
    </li>
  ));

  return <ul>{villagerElements}</ul>;
};

VillagerList.propTypes = {
  villagers: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      species: PropTypes.string.isRequired,
      phrase: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default VillagerList;
