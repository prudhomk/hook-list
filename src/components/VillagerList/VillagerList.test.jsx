/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerList from './VillagerList';
import { MemoryRouter } from 'react-router-dom';

describe('Villager List component', () => {
  const list = [
    {
      name: 'hello',
      species: 'phrase',
      phrase: 'species',
      image: 'whatever',
      _id: 'string'
    }
  ];
  afterEach(() => cleanup());
  it('renders Villager List', () => {
    const { asFragment } = render(<MemoryRouter><VillagerList villagers={list}/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
