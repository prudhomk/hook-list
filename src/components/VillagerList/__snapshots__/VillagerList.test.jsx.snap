// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager List component renders Villager List 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/string"
      >
        <figure>
          <img
            alt="hello"
            src="whatever"
          />
          <figcaption>
            <p>
              hello
            </p>
            <p>
              phrase
            </p>
            <p>
              species
            </p>
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
